packages/react-reconciler/src/ReactFiberHooks.old.js

```diff
@@ -1777,6 +1780,52 @@ function rerenderOpaqueIdentifier(): OpaqueIDType | void {
   return id;
 }
 
+function mountRefresh() {
+  const cacheInstance: CacheInstance = readContext(CacheContext);
+  return mountCallback(refreshCache.bind(null, cacheInstance), [cacheInstance]);
```

Comment:
It doesn't seem to me that these should need to be invalidated. It should always be the same callback instance.

Instead this should close over the nearest provider which can't change. If that's not true and this is sometimes a higher provider then that seems like a bug because it should always refresh the nearest specified as the user specified.

Reply:
> it should always refresh the nearest specified as the user specified.

I thought we still weren't sure if it would refresh the nearest provider or the root of the "There" boundary. Right now it's always the latter. Kinda feel like you want both abilities. But for now I will change it to be the local one, though.